### DISCUSSION TITLE
Monorepo & /play fork

### DISCUSSION THREAD

**Discussion by tmcw** [placemark/placemark#52] [Announcements]
This repo is now a monorepo. It uses [pnpm](https://pnpm.io/) to do it. In effect, this lets multiple npm modules with multiple package.json files live in the same repository and reuse some of each other's dependencies.

The big change that this enabled is #51 - a fork that supports the Placemark Play free interface again. And that interface is now hosted at https://play.placemark.io/ and the converter is online at https://play.placemark.io/converter 

Both are client-side-only applications, so I can run them cheaply. It would be great if Placemark was modularized and this was a different configuration or a minimal frontend that reused modules, but this is an in-between step which is pretty dirty: the `packages/play` codebase is basically placemark with a ton of deletions. I have just enough time to try to make something useful, not enough time to do a proper job making it modularized right now. If you have time, please submit a PR!

<hr /><em>This discussion was created from the release <a href='https://github.com/placemark/placemark/releases/tag/v2.0.0'>Monorepo & /play fork</a>.</em>

**Comment by davenquinn** [comment]
Hey @tmcw, I've enjoyed watching Placemark evolve and am excited to explore its innards and potentially harness of your innovations towards my own ends.

I'm running Macrostrat and am fairly involved in several other geological apps. For the moment, we've got significant resources towards improving web-based workflows around geologic mapping (including adding some editing capabilities to our platforms). We'd probably have some cycles to spare to contribute to modularizing/testing some of Placemark's client-side code. That said

- Only some of the components will likely be targets for us
- We'd presumably have to modify some of them or make them more configurable to fit comfortably within our environment
- We lack the full overview or experience to know what the best way to package/bundle/release web component libraries in a way that fits the goals of the project.

If you or others with more of that experience could set or sketch the broad direction here, we could probably contribute some PRs to flesh things out. It'd be nice to make use of this without contributing too much to fragmentation.

**Reply by tmcw** [reply]
Hey Daven - I'm a _huge_ fan of Macrostat, it's one of the coolest applications on the internet!

It really depends on the component - Placemark has a lot of interdependencies in some components, because it tends to use a few utility libraries like `purify-ts` and `ts-pattern`, as well as lots of shared types. So it may be tricky to extract certain bits of code and make them more general-purpose. Maybe once you have a specific part to pull out in mind you can email me and we can chat or I'll prototype a componentized version?
